app/qml/map/MapWrapper.qml

```diff
@@ -0,0 +1,534 @@
+  }
+
+  function processRecordedPair( pair ) {
+    if ( _digitizingController.isPairValid( pair ) ) {
+      root.recordingFinished( pair )
+    }
+    else {
+      root.notify( qsTr( "Recorded feature is not valid" ) )
+      root.recordingCanceled()
+    }
+    root.state = "View"
+  }
+
+  function addRecordPoint() {
+    let recordedPoint = findRecordedPoint()
+
+    let isPointGeometry = _digitizingController.hasPointGeometry( _digitizingController.layer )
+    let isUsingGPS = _digitizingController.useGpsPoint
+    let hasAssignedValidPair = root.featurePairToEdit && root.featurePairToEdit.valid
+
+    if ( root.state === "RecordFeature" ) {
+      if ( isPointGeometry ) {
+        let newPair = _digitizingController.pointFeatureFromPoint( recordedPoint, isUsingGPS )
+        processRecordedPair( newPair )
+        return
+      }
+
+      if ( !_digitizingController.recording )
+        _digitizingController.startRecording()
+
+      _digitizingController.addRecordPoint( recordedPoint, isUsingGPS )
+    }
+    else if ( root.state === "EditGeometry" ) {
+      if ( isPointGeometry && hasAssignedValidPair ) {
+        let changed = _digitizingController.changePointGeometry( root.featurePairToEdit, recordedPoint, isUsingGPS )
+        root.editingFinished( changed )
+        return
+      }
+    }
+    else if ( root.state === "AddGeometry" ) {
+      if ( isPointGeometry ) {
+        if ( !hasAssignedValidPair ) return
+
+        let changed = _digitizingController.changePointGeometry( root.featurePairToEdit, recordedPoint, isUsingGPS )
+        root.addingGeometryFinished( changed )
+        return
+      }
+
+      // adding line/polygon geometry
+      _digitizingController.addRecordPoint( recordedPoint, isUsingGPS )
+    }
+  }
+
+  function isPositionOutOfExtent() {
+    let border = InputStyle.mapOutOfExtentBorder
+    return ( ( _positionKit.screenPosition.x < border ) ||
+            ( _positionKit.screenPosition.y < border ) ||
+            ( _positionKit.screenPosition.x > _map.width - border ) ||
+            ( _positionKit.screenPosition.y > _map.height - border )
+            )
+  }
+
+  function updatePosition() {
+    let autoCenterDuringRecording = _digitizingController.useGpsPoint && root.isInRecordState
+    let autoCenterDuringViewing = !root.isInRecordState && __appSettings.autoCenterMapChecked && isPositionOutOfExtent()
+
+    if ( autoCenterDuringRecording || autoCenterDuringViewing ) {
+      let useGpsPoint = _digitizingController.useGpsPoint
+      centerToPosition()
+    }
+
+    _digitizingHighlight.positionChanged()
+  }
+
+  function clear() {
+    // clear all previous references to old project (if we don't clear references to the previous project,
+    // highlights may end up with dangling pointers to map layers and cause crashes)
+    _highlightIdentified.featureLayerPair = null
+    _digitizingHighlight.featureLayerPair = null
+  }
+
+  states: [
+    State {
+      name: "View"
```

Comment:
We used to name states starting with small letter as well as can be seen [here. in coding conventions](https://doc-snapshots.qt.io/qt6-dev/qml-codingconventions.html).
Any particular reason while states starts with a capital letter?